apluslms_roman/tests/test_cli.py

```diff
@@ -0,0 +1,285 @@
+import sys
+
+from contextlib import contextmanager
+from io import StringIO
+from os import chdir, getcwd, remove
+from os.path import join, exists, isdir
+from shutil import rmtree
+from unittest import mock, TestCase
+
+from apluslms_roman import cli
+
+
+@contextmanager
+def capture_output():
+    new_out, new_err = StringIO(), StringIO()
+    old_out = (sys.stdout, cli.stdout)
+    old_err = (sys.stderr, cli.stderr)
+    try:
+        sys.stdout = cli.stdout = new_out
+        sys.stderr = cli.stderr = new_err
+        yield (sys.stdout), (sys.stderr)
+    finally:
+        sys.stdout, sys.stderr = old_out[0], old_err[0]
+        cli.stdout, cli.stderr = old_out[1], old_err[1]
+
+COURSE = join(getcwd(), 'mock_course')
+CONFIG = 'roman.yml'
+
+def delete(file_):
+    if exists(file_):
+        if isdir(file_):
+            rmtree(file_)
+        else:
+            remove(file_)
+
+def use_temp_file(func):
+    def wrapper(*args):
+        delete('temp.yml')
+        command_test(args[0], "roman -f temp.yml init")
+        try:
+            func(*args)
+            delete('temp.yml')
+        except Exception:
+            delete('temp.yml')
+            raise
+    return wrapper
+
+def add_hello(func):
+    def wrapper(*args):
+        command_test(args[0], "roman -f temp.yml step add hello-world --name hello")
+        func(*args)
+    return wrapper
+
+def format_output(io_string):
+    return io_string.getvalue().strip()
+
+def command_test(cls, command, exit_code=None):
+    if isinstance(command, str):
+        command = command.split(" ")
+    sys.argv = command
+    exit_code = str(exit_code) if exit_code is not None else '0'
+    try:
+        with capture_output() as (out, err):
+            cls.assertRaisesRegex(SystemExit, exit_code, cli.main)
```

Comment:
`cli.main` could take `argv` as an argument and if None is given, then it would use `sys.argv`.

Reply:
so `args = parser.parse_args()` is the part that should be changed?